examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/CastingServerBridge.mm

```diff
@@ -118,10 +109,157 @@ - (instancetype)init
+    // take a callback to indicate that the low-level SDK operation has been initiated (the "started"
+    // callback), and a separate set of result callbacks to be invoked when the low-level SDK operation
+    // has been completed (the "result" callbacks).
+    //
+    // All of the work is serialized on the _chipWorkQueue, with the started callback being invoked after the
+    // Matter SDK is called with the result callbacks. This poses a challenge because we don't have a strong
+    // guarantee that the Matter SDK will not complete its work and invoke the result callback before control
+    // is returning to the _chipWorkQueue thread; this can cause callbacks to be received in an unexpected
+    // order, where the result might arrive before the client knows that the call was started.
+    //
+    // To avoid this confusion, we perform a "double async" dispatch to ensure that events are delivered in a
+    // sensible order. First, we schedule the result callback to run on the _chipWorkQueue (so it *must* run
+    // after the end of the CastingServerBridge's wrapper logic that already started on the _chipWorkQueue)
+    // and then immediately, asynchronously dispatch the callback onto the provided client queue.
+    dispatch_async(_chipWorkQueue, ^{
+        if (nil != description) {
+            ChipLogProgress(AppServer, "[ASYNC (CHIP)] CastingServerBridge dispatching %s", [description UTF8String]);
+        }
+        dispatch_async(queue, ^{
+            if (nil != description) {
+                ChipLogProgress(AppServer, "[ASYNC (CLIENT)] CastingServerBridge BEGIN %s", [description UTF8String]);
+            }
+            block();
+            if (nil != description) {
+                ChipLogProgress(AppServer, "[ASYNC (CLIENT)] CastingServerBridge END %s", [description UTF8String]);
+            }
+        });
+    });
+}
+
+/**
+ A utility function that invokes an Objective C Block on the Matter Dispatch Queue in a thread-safe manner.
+
+ The result of the Objective C Block will be relayed to the `blockCompleteCallback`.
+
+ @param description  Optional. A description of the Objective C Block to be invoked for diagnostic purposes.
+ @param blockCompleteCallback The callback to be invoked with the result of the Objective C Block.
+ @param callbackQueue The Dispatch Queue on which the callback shall be invoked.
+ @param block The Objective C Block to dispatch.
+ */
+- (void) withCastingServerInvokeBlock:(const NSString * _Nullable)description
+               withCompletionCallback:(void (^_Nonnull)(bool))blockCompleteCallback
+                              onQueue:(dispatch_queue_t)callbackQueue
+                                block:(CHIP_ERROR (^_Nonnull)(CastingServer *)) block
+{
+    [self dispatchOnMatterSDKQueue:description block:^{
+        CastingServer * castingServer = CastingServer::GetInstance();
+        const CHIP_ERROR result = block(castingServer);
```

Comment:
nit, might be useful to add a message here indicating why we're using a block at all for this object
```suggestion
        // Create a block because it is being used in the dispatch
        const CHIP_ERROR result = block(castingServer);
```

Reply:
Done